File: lihua-biz/lihua-system/src/main/java/com/lihua/entity/SysRole.java
package com.lihua.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.lihua.mybatis.model.BaseEntity;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.ArrayList;
import java.util.List;

@EqualsAndHashCode(callSuper = true)
@Data
public class SysRole extends BaseEntity {
    /**
     * 主键
     */
    @TableId(type = IdType.ASSIGN_ID)
    private String id;

    /**
     * 角色名称
     */
    @NotNull(message = "请输入角色名称")
    private String name;

    /**
     * 角色编码
     */
    @NotNull(message = "请输入角色编码")
    private String code;

    /**
     * 角色状态
     */
    @NotNull(message = "请选择角色状态")
    private String status;

    /**
     * 备注
     */
    private String remark;

    /**
     * 菜单id集合
     */
    @TableField(exist = false)
    private List<String> menuIds = new ArrayList<>();
}

File: lihua-app/src/api/system/role/type/sys-role.ts
export interface SysRole {
  /**
   * 主键
   */
  id?: string;
  /**
   * 角色名称
   */
  name?: string;
  /**
   * 角色编码
   */
  code?: string;
  /**
   * 角色状态
   */
  status?: string;
  /**
   * 备注
   */
  remark?: string;
  /**
   * 菜单id集合
   */
  menuIds?: string[] | { checked: string[] };
}